tools/gen/impl_template

```diff
@@ -305,6 +305,39 @@ fail:
 	fromwire_fail(cursor, max);
 	return false;
 }
+bool ${tlv.name}_is_valid(const struct ${tlv.struct_name()} *record)
```

Comment:
Please have this return the problematic offset within the TLV? There's a FIXME for this requirement in the onion parsing code, where we're supposed to return the offset which was the problem in the error msg.

Reply:
We can return the exact offset of the offending field in `fromwire_tlv_payload` which has the locations, while here we would have to track the offset by accumulating the `varint` lengths and field lengths while verifying. I can add the offset in the `fromwire` code and just add the field index that we get upset about if you'd like :-)

Reply:
Might make more sense anyway, since we can then say things like "Unknown even type x in field y" instead of saying I got upset somewhere at byte `z`.